packages/core/test/system.js

```diff
@@ -258,18 +258,52 @@ test('sorts media queries when responsive object values are used', () => {
         md: '40em',
         lg: '64em',
         xl: '128em',
-      }
+      },
     },
     padding: { _: 16, lg: 64, xl: 128 },
-    margin: { sm: 4, md: 8 },
-    color: { lg: 'tomato' },
+    margin: { _: 0, sm: 4, md: 8 },
+    color: { _: 'olive', lg: 'tomato' },
+  })
+  expect(Object.keys(styles)).toEqual([
+    '@media screen and (min-width: 32em)',
+    '@media screen and (min-width: 40em)',
+    '@media screen and (min-width: 64em)',
+    '@media screen and (min-width: 128em)',
+    'padding',
+    'margin',
+    'color',
+  ])
+})
+
+test('sorts media queries when both responsive object and array values are allowed', () => {
```

Comment:
Oh this is meant to test the logic on the line with `cache.breakpointHasKeys` - that we do end up sorting if both arrays and objects are possible as breakpoints.

Reply:
Happy to reword this for clarity! Any suggestions?

Maybe 
`sorts media queries when both object and array breakpoint values are expected`

Reply:
Ah, makes sense. Does this still work when mixing array and plain object values? That's mostly what I was curious about, since that's how I interpreted the description here

Reply:
I believe they do - I'll update this test to cover that case